provisionersdk/logger.go

```diff
@@ -0,0 +1,71 @@
+package provisionersdk
+
+import (
+	"bufio"
+	"io"
+
+	"github.com/coder/coder/provisionersdk/proto"
+)
+
+type Logger interface {
+	Log(l *proto.Log) error
+}
```

Comment:
Is this interface primarily to merge the `Provision` and `Parse` log streams? I think abstracting this makes it confusing to follow the code, and just to save ~30 lines of code.

Reply:
Yeah, the idea is to be able to write generic logging code, like `LogWriter()` that work on either kind of stream.  This is something I found really useful when I prototyped my docker-compose provisioner, so I think it's really useful to have generic logging routines.